packages/btcindexer/src/btcindexer.ts

```diff
@@ -460,6 +496,49 @@ export class Indexer {
 		}
 	}
 
+	async detectMintedReorgs(blockHeight: number): Promise<void> {
+		logger.debug({
+			msg: "Checking for reorgs on minted transactions",
+			method: "detectMintedReorgs",
+			blockHeight,
+		});
+
+		const mintedTxs = await this.storage.getMintedTxs(blockHeight);
+		if (!mintedTxs || mintedTxs.length === 0) {
+			return;
+		}
+
+		for (const tx of mintedTxs) {
+			try {
+				const currentBlockHash = await this.storage.getBlockHash(
+					tx.block_height,
+					tx.btc_network,
+				);
+
```

Comment:
you should redo this loop - I think you can do it in a single SQL to join btc_blocks with nbtc_minting, and check if hash is different.